SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/EndToEndTests.java

```diff
@@ -46,6 +47,20 @@ public void overSumTest() {
         this.testQuery(query, new DBSPZSetLiteral(t, t));
     }
 
+    @Test
+    public void testConcat() {
+        String query = "SELECT T.COL4 || ' ' || T.COL4 FROM T";
+        DBSPExpression lit = new DBSPTupleExpression(new DBSPStringLiteral("Hi Hi"));
+        this.testQuery(query, new DBSPZSetLiteral(lit, lit));
+    }
+
+    @Test
+    public void testConcatNull() {
+        String query = "SELECT T.COL4 || NULL FROM T";
+        DBSPExpression lit = new DBSPTupleExpression(DBSPLiteral.none(DBSPTypeString.instance.setMayBeNull(true)));
+        this.testQuery(query, new DBSPZSetLiteral(lit, lit));
```

Comment:
What are these test asserting on?

Reply:
They check the produced output. testQuery takes a query and an expected output and generates Rust which asserts that the two are equal.